pkg/cmd/cli/describe/deployments.go

```diff
@@ -93,24 +106,23 @@ func (d *DeploymentConfigDescriber) Describe(namespace, name string, settings kc
 			}
 		} else {
 			header := fmt.Sprintf("Deployment #%d (latest)", deployutil.DeploymentVersionFor(deployment))
-			printDeploymentRc(deployment, d.kubeClient, out, header, true)
+			// Show details if the current deployment is the active one or it is the
+			// initial deployment.
+			printDeploymentRc(deployment, d.kubeClient, out, header, (deployment.Name == activeDeploymentName) || len(deploymentsHistory) == 0)
```

Comment:
Actually making two different calls to the api server is not consistent. How about looking into the list for the latest deployment instead of Getting it?